docs/End_to_End_Testing.md

```diff
@@ -0,0 +1,74 @@
+# End-to-End Testing
+
+End-to-end test must be included as a part of a PR for any contribution to Flux including: 
+[Flux Functions](/Flux_Functions.md), 
+[Standalone Scalar Functions](/Scalar_Functions.md), 
+[Sink/Source Functions](/Source_Sink_Functions.md), 
+[Stream Transformation Functions](/Stream_Transformation_Functions.md). 
+
+## Required guidelines
+
+Please help us make the contribution process easier by providing feedback about your experience and any technical hurdles you encountered here.
+
+### Pure Flux Code Function Test Guidelines
+
+- The file name must end with _test.flux
+- Tests are now adjacent to their go files. For examples [columns_test.flux](../stdlib/universe/columns_test.flux) is adjacent to [columns.go](./stdlib/universe/columns.gp)
+- The package must be named after the containing folder. For example, in [universe](../stdlib/universe/), the test must start with package universe_test. 
+- Also, remember to ```make generate``` before adding files to the commit.
+
+Example Test:
+```
+package universe_test
+ 
+import "testing"
+
+```
+
+- Set this option to a distant future so that queries based on ```now()``` will be consistent
+```option now = () => (2030-01-01T00:00:00Z)```

```

Comment:
I'd reformat to obtain:

```
package universe_test
 
import "testing"
```

 - Fix the time of execution for consistent results across multiple test runs:

```
option now = () => (2030-01-01T00:00:00Z)
````

 - ...